src/pharmpy/tools/structsearch/pkpd.py

```diff
@@ -95,48 +95,51 @@ def create_pkpd_models(
         elif modeltype == 'indirect':
             for argument in list(product(pd_types, [True, False])):
                 pkpd_model = add_indirect_effect(model, *argument)
-                pkpd_model = pkpd_model.replace(
-                    description=f"{modeltype}_{argument[0]}_prod={argument[1]}"
-                )
+                if argument[1]:
+                    pkpd_model = pkpd_model.replace(
+                        description=f"{modeltype}_{argument[0]}_production"
+                    )
+                else:
+                    pkpd_model = pkpd_model.replace(
+                        description=f"{modeltype}_{argument[0]}_degradation"
+                    )
                 models.append(pkpd_model)
 
-        final_models = []
-        index = 1
-        for pkpd_model in models:
-            pkpd_model = set_name(pkpd_model, f"structsearch_run{index}")
-            index += 1
-
-            # initial values
-            if b_init is not None:
-                pkpd_model = set_initial_estimates(pkpd_model, b_init)
-            if ests is not None:
-                pkpd_model = fix_parameters_to(pkpd_model, ests)
-
-            if emax_init is not None and ec50_init is not None:
-                pkpd_model = set_initial_estimates(pkpd_model, {'POP_E_MAX': emax_init})
-                pkpd_model = set_initial_estimates(pkpd_model, {'POP_EC_50': ec50_init})
-                pkpd_model = set_initial_estimates(pkpd_model, {'POP_SLOPE': emax_init / ec50_init})
-            if mat_init is not None:
-                pkpd_model = set_initial_estimates(
-                    pkpd_model, {'POP_KE0': 1 / mat_init, 'POP_K_OUT': 1 / mat_init}
-                )
-
-            pkpd_model = unconstrain_parameters(pkpd_model, ['SLOPE'])
-            pkpd_model = set_lower_bounds(pkpd_model, {'POP_E_MAX': -1})
-
-            # set iiv
-            for parameter in ["B", "SLOPE"]:
-                try:
-                    pkpd_model = add_iiv(pkpd_model, [parameter], "exp")
-                except ValueError:
-                    pass
+    final_models = []
+    index = 1
+    for pkpd_model in models:
```

Comment:
Since your index increases by 1 in each loop, you could use `for i, pkpd_model in enumerate(models, 1):` instead of setting the index each loop